gapic/samplegen/samplegen.py

```diff
@@ -511,3 +512,59 @@ def generate_sample(sample,
                                                 imports=[],
                                                 calling_form=calling_form,
                                                 calling_form_enum=utils.CallingForm)
+
+
+def generate_manifest(fpaths_and_samples,
+                      api_schema,
+                      *,
+                      manifest_time: int = None):
+    """Generate a samplegen manifest for use by sampletest
+
+    Args:
+        fpaths_and_samples (Iterable[Tuple[str, Mapping[str, ?]]]):
+                         The file paths and samples to be listed in the manifest
+
+        api_schema (~api.API): An API schema object.
+        manifest_time (int): Optional. An override for the timestamp in the name of the manifest filename.
+                             Primarily used for testing.
+
+    Returns:
+        Tuple[str, Dict[str,?]]: The filename of the manifest and the manifest data as a dictionary.
+
+    """
+    python_env = {
+        "environment": "python",
+        "bin": "python3",
+        "base_path": "sample/base/directory",
+        "invocation": "{bin} {path} @args",
+    }
+    all_info = {
+        "type": "manifest/samples",
+        "schema_version": 3,
+        "python": python_env,
```

Comment:
I assume you have this so it can be an anchor that is referenced by each individual sample. That is correct.

Reply:
That's the plan.